src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/LanguageServer.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using StreamJsonRpc;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public abstract class LanguageServer<RequestContextType> : ILanguageServer
+{
+    protected readonly ILspLogger _logger;
+
+    protected readonly string _serverKind;
+
+    public bool IsInitialized { get; private set; } = false;
+
+    public bool HasShutdownStarted => _shuttingDown;
```

Comment:
confused where `_shuttingDown` is defined.  i don't see it.  also concerned about raceyness here.

Reply:
_shuttingDown was a private member so I removed it, I can clean this up when I come with the rest of my updates from the feedback here.